backend/common/src/main/java/ai/verta/modeldb/common/futures/FutureRest.java

```diff
@@ -0,0 +1,71 @@
+package ai.verta.modeldb.common.futures;
+
+import io.grpc.Context;
+import io.opentracing.util.GlobalTracer;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ForkJoinPool;
+import org.springframework.lang.NonNull;
+
+@SuppressWarnings({"squid:S100"})
+public class FutureRest {
+  private FutureRest() {}
+
+  // Injects the result of the future into the grpc StreamObserver as the return of the server
+  public static <T> CompletableFuture<T> serverResponse(InternalFuture<T> f, Executor ex) {
+    CompletableFuture<T> promise = new CompletableFuture<>();
+    f.whenComplete(
+        (v, t) -> {
+          if (t == null) {
+            try {
+              promise.complete(v);
+            } catch (Throwable e) {
+              promise.completeExceptionally(e);
+            }
+          } else {
+            promise.completeExceptionally(t);
+          }
+        },
+        ex);
+    return promise;
+  }
```

Comment:
Let's just add `toCompletableFuture` to the internal future functionality. We don't need this more complex wrapping, right?

Reply:
No, we need to handle error code and exception handling at our end so this will improve with it at other PR https://github.com/VertaAI/modeldb/pull/3033/files#diff-56cedd6e9fc5cded11b6d5a79cca7ab71f4a9c892a8d06d081bcdbc2c8d509deL23

Reply:
Asking people to review a PR with consideration for all the possible PRs that come after that defeats the purpose of the PR change itself. Let's stop doing that, please. It makes review _very_ painful and slow.